@here/harp-mapview/lib/PickHandler.ts

```diff
@@ -149,23 +149,80 @@ export class PickHandler {
             if (pickListener.done && pickListener.furthestResult!.distance < distance) {
                 // Stop when the listener has all results it needs and remaining tiles are further
                 // away than then furthest pick result found so far.
                 break;
             }
-            intersects.length = 0;
-            rayCaster.intersectObjects(tile.objects, true, intersects);
-            for (const intersect of intersects) {
-                pickListener.addResult(this.createResult(intersect));
-            }
+            this.intersectObjects(tile, intersects, pickListener, rayCaster);
+            this.intersectDependentObjects(
+                tile,
+                intersects,
+                pickListener,
+                rayCaster,
+                checkedDependencies
+            );
         }
 
         pickListener.finish();
         return pickListener.results;
     }
 
+    // Adds the list of intersection results to the pick listener. The intersection results are
+    // cleared after being added.
+    private addIntersectResults(intersects: THREE.Intersection[], pickListener: PickListener) {
+        for (const intersect of intersects) {
+            pickListener.addResult(this.createResult(intersect));
+        }
+        intersects.length = 0;
+    }
+
+    // Intersects the tiles objects using the supplied raycaster and adds the intersection results
+    // to the PickListener
+    private intersectObjects(
+        tile: Tile,
+        intersects: THREE.Intersection[],
+        pickListener: PickListener,
+        rayCaster: THREE.Raycaster
+    ) {
+        rayCaster.intersectObjects(tile.objects, true, intersects);
+        this.addIntersectResults(intersects, pickListener);
+    }
+
+    // Intersects the dependent tile objects using the supplied raycaster and adds the intersection
+    // results to the PickListener. Note, because multiple tiles can point to the same dependency
+    // we need to store which results we have already raycasted, see checkedDependencies.
+    private intersectDependentObjects(
+        tile: Tile,
+        intersects: THREE.Intersection[],
+        pickListener: PickListener,
+        rayCaster: THREE.Raycaster,
+        checkedDependencies: Set<number>
+    ) {
+        for (const tileKey of tile.dependencies) {
+            const mortonCode = tileKey.mortonCode();
+            if (checkedDependencies.has(mortonCode)) {
+                continue;
+            }
+            checkedDependencies.add(mortonCode);
+            const otherTile = this.mapView.visibleTileSet.getCachedTile(
+                tile.dataSource,
+                tileKey,
+                tile.offset,
+                this.mapView.frameNumber
+            );
+            if (otherTile !== undefined) {
+                rayCaster.intersectObjects(otherTile.objects, true, intersects);
+            }
+        }
+        this.addIntersectResults(intersects, pickListener);
```

Comment:
better do this once per iteration of the loop at intersectMapObjects(). It's enough to do it just before processing the next tile.